xarray/core/missing.py

```diff
@@ -48,7 +48,7 @@ def _get_nan_block_lengths(
         .where(valid)
         .bfill(dim=dim)
         .where(~valid, 0)
-        .fillna(index[-1] - valid_arange.max())
+        .fillna(index[-1] - valid_arange.max(dim=dim))
```

Comment:
These types of problems are always tricky.
Probably best if you use `dim=[dim]`, this should work for non-str dimensions.
This should fix the mypy issues

Reply:
Thank you! I saw the issue, but was not sure how to resolve it correctly.